onecodex/viz/_heatmap.py

```diff
@@ -238,6 +251,24 @@ def plot_heatmap(
                 haxis, header=alt.Header(titleOrient="bottom", labelOrient="bottom")
             )
 
+        # Drop all-NaN rows, convert remaining NaNs to 0s, and concat
+        # dropped all-NaN rows to the end
+        dropped = []
+        for classification_id in all_nan_classification_ids:
+            d = df[(df == classification_id).any(axis=1)]
+            if self.__class__.__name__ == "OneCodexAccessor":
+                # the df of a `OneCodexAccessor` does not have NaNs, and we want to display
+                # all_nan_classification_id columns as white instead of color corresponding to 0
+                d = d.replace(0.0, np.nan)
+            dropped.append(d)
+            index = df[(df == classification_id).any(axis=1)].index
+            df = df.drop(index)
+
+        df = df.replace(np.nan, 0)
+        df = pd.concat([df, *dropped])
+
+        assert set(df["Label"].values) == set(labels_in_order)
```

Comment:
is this assert needed?

Reply:
Not strictly - @audy had suggested it to make sure that we aren't ever ending up in a situation where we have a mismatch between the chart and the `labels_in_order`, which would lead to an incorrect plot